ballerina/tests/producer_client_tests.bal

```diff
@@ -304,6 +304,62 @@ function saslProducerTest() returns error? {
     check consumer->close();
 }
 
+@test:Config{enable: true}
+function saslScram256ProducerTest() returns error? {
+    string topic = "sasl-scram-256-producer-test-topic";
+    kafkaTopics.push(topic);
+
+    ProducerConfiguration producerConfigs = {
+        clientId: "test-producer-06",
+        auth: authScram256Config,
+        securityProtocol: PROTOCOL_SASL_PLAINTEXT
+    };
+
+    Producer kafkaProducer = check new (SASL_URL, producerConfigs);
+    Error? result = kafkaProducer->send({topic: topic, value: TEST_MESSAGE.toBytes() });
+    test:assertFalse(result is error, result is error ? result.toString() : result.toString());
```

Comment:
Any reason to follow this structure ? We could have done this IMO
```suggestion
    check kafkaProducer->send({topic: topic, value: TEST_MESSAGE.toBytes() });
```

Reply:
Will attend to these test case refactors separately